v3/src/components/data-display/models/data-configuration-model.ts

```diff
@@ -135,6 +138,8 @@ export const DataConfigurationModel = types
   }))
   .views(self => ({
     filterCase(data: IDataSet, caseID: string, caseArrayNumber?: number) {
+      // If the case is hidden we don't plot it
+      if (self.hiddenCases.includes(caseID)) return false
```

Comment:
This is an O(n^2) lookup which could become a problem with large data sets. Could have a view method which returned a  set to get O(n) lookups, for instance.